docs/questions.rst

```diff
@@ -419,3 +419,37 @@ tmt example plan (L2 metadata):
 
 See the :ref:`/spec/plans/discover/fmf` plugin documentation for
 more details.
+
+
+Migrating provision.fmf
+------------------------------------------------------------------
+
+The ``provision.fmf`` file is used to specify storage and network devices.
+In this migration, the contents of the ``provision.fmf`` file is moved to the
+``provision`` step under ``hardware`` specification.
+
+``provision.fmf`` example:
+
+.. code-block:: yaml
+
+    standard-inventory-qcow2:
+        qemu:
+            drive:
+            - size: 10737418240
+            - size: 10737418240
+            - size: 10737418240
+
+tmt example plan (L2 metadata):
+
+.. code-block:: yaml
+
+    provision:
+        how: virtual
+        hardware:
+            disk:
+                - size: ">10GiB"
+                - size: ">10GiB"
+                - size: ">10GiB"
+
```

Comment:
@janhavlin could we mention that this currently works only with Artemis provisioner, so really it will work only in Testing Farm?

Reply:
Sure, mentioning it sounds good. Just a question:

> this currently works only with Artemis provisioner

This part sounds right.

> so really it will work only in Testing Farm?

Is this really true? In the [`disk.fmf`](https://github.com/teemtee/tmt/blob/main/spec/hardware/disk.fmf) spec, it is mentioned that it is implemented by `tmt/steps/provision/artemis.py`, therefore it should work with both Testing Farm and `artemis` TMT plugin when using `tmt` directly?

Reply:
I wouldn't mention Artemis by name. The support will change, this feature will be supported by more plugins, and nobody will recall updating this particular spot to include them. I believe the `See ...` note below is the right way, refer to a page dedicated to HW requirements, and we can extend it with "... for more details and plugin support.", for example.

Reply:
> I wouldn't mention Artemis by name. The support will change, this feature will be supported by more plugins, and nobody will recall updating this particular spot to include them. I believe the `See ...` note below is the right way, refer to a page dedicated to HW requirements, and we can extend it with "... for more details and plugin support.", for example.

Edit: I wouldn't mention TF by name either...

Reply:
@happz @janhavlin so from discussion, let's link from this guide to the Testing Farm `Test Request` page:

https://docs.testing-farm.io/general/0.1/test-request.html

Reply:
Rather than adding note about that this is supported only there.

Reply:
Added a link to the TF docs. PTAL. Although I'm not sure how to link the TF docs if naming TF is not a good practice.

Reply:
Mentioning TF in *specs* would be sort of vendor-lock-in-ish, it's always good to think whether it makes sense to point out a single tmt runner in such case. But here we are in a "how to migrate" part of tmt docs, and here it makes more sense as we are dealing with existing systems. No worries here.

I'd just maybe reword the sentence a bit: it's definitely possible to express the requirement, TF is mighty and can do anything, how about aligning it with the migration effort? Let's say "If you were using `provision.yaml` with Testing Farm, check out the TF docs on this HW requirement, https://..., for more details and how TF works with tmt metadata." I don't like the notion of "hey, this is possible with TF" - yep, that's true, but it will be possible with other systems one day. But when looking at it from the PoV of folks who were using the file, and want to migrate and use TF, which was the original motivation, singling TF out seems more justifiable to me.

Reply:
I see, that is indeed a bit better wording. One issue that I see with it is that currently, the TF docs page (https://docs.testing-farm.io/general/0.1/test-request.html) is a bit lacking with the explanation of "how TF works with tmt metadata" since currently it only describes how to express the same hardware specification with TF request syntax. That is also something that needs to improve as we already discussed it?

Reply:
I guess so, yes, on TF side.

Reply:
@janhavlin waiting for docs update on TF side and resolving this one.